pkg/api/types.go

```diff
@@ -1192,13 +1195,15 @@ type ServiceSpec struct {
 	// None can be specified for headless services when proxying is not required
 	ClusterIP string `json:"clusterIP,omitempty"`
 
-	// Type determines how the service will be exposed.  Valid options: ClusterIP, NodePort, LoadBalancer
-	Type ServiceType `json:"type,omitempty"`
 
 	// ExternalIPs are used by external load balancers, or can be set by
 	// users to handle external traffic that arrives at a node.
 	ExternalIPs []string `json:"externalIPs,omitempty"`
 
+	// Only applies to Service Type: LoadBalancer
+	// LoadBalancer will get created with the IP specified in this field.
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
```

Comment:
Right .. so before this change, what value turned up in `TCPLoadBalancer.EnsureTCPLoadBalancer(... externalIP ..)` ?  I think I've already implemented that for OpenStack as you're describing for `LoadBalancerIP`.

I guess I just never thought about where that value came from before.  From reading servicecontroller.go(*), it seems this *is* `ExternalIPs`.  So again I'm back to "What's new here? How is `LoadBalancerIP` different to `ExternalIPs` (with type=LoadBalancer)?"

(*) https://github.com/kubernetes/kubernetes/blob/cd798a471f210ad824cbd0d5dcd7f41af104684c/pkg/controller/service/servicecontroller.go#L385